dap-variables.el

```diff
@@ -0,0 +1,187 @@
+;;; dap-variables.el --- Launch configuration variables -*- lexical-binding: t -*-
```

Comment:
Does it make sense to move this file into `dap-utils`?
This file is mostly utilities too.

Reply:
Perhaps the functions could be included in dap-utils? But the purpose of the file is to define variables that can be expanded in launch configurations, and all the functions serve this purpose. That is, they implement variables that can be expanded.
By variables I mean that if you have a launch configuration like this (taken from my gtk-print project)
```json
{
    "configurations": [
        {
            "name": "Debug",
            "type": "gdb",
            "request": "launch",
            "target": "${workspaceFolder}/build/gtk-print",
            "debugger_args": [ "-ex", "set args /tmp/printme.pdf" ]
        },
        {
            "name": "Debug (attach to localhost:4596)",
            "type": "gdb",
            "request": "attach",
            "executable": "${workspaceFolder}/build/gtk-print",
            "target": ":4596",
            "remote": true
        }
    ]
}
```
Things like ${workspaceFolder} would be expanded by calling the appropriate functions in dap-variable-launch-configuration-variables. This works in normal launch configurations in lisp form too.